autogluon/task/tabular_prediction/tabular_prediction.py

```diff
@@ -444,18 +524,39 @@ def fit(train_data, label, tuning_data=None, output_directory=None, problem_type
         learner.fit(X=train_data, X_test=tuning_data, scheduler_options=scheduler_options,
                     hyperparameter_tune=hyperparameter_tune, feature_prune=feature_prune,
                     holdout_frac=holdout_frac, num_bagging_folds=num_bagging_folds, num_bagging_sets=num_bagging_sets, stack_ensemble_levels=stack_ensemble_levels,
-                    hyperparameters=hyperparameters, time_limit=time_limits_orig, save_data=cache_data, verbosity=verbosity)
+                    hyperparameters=hyperparameters, time_limit=time_limits_orig, save_data=cache_data, save_bagged_folds=save_bagged_folds, verbosity=verbosity)
+
+        predictor = TabularPredictor(learner=learner)
+
+        keep_only_best = kwargs.get('keep_only_best', False)
+        if refit_full is True:
+            if keep_only_best is True:
+                if set_best_to_refit_full is True:
+                    refit_full = 'best'
+                else:
+                    logger.warning(f'refit_full was set to {refit_full}, but keep_only_best=True and set_best_to_refit_full=False. Disabling refit_full to avoid training models which would be automatically deleted.')
+                    refit_full = False
+            else:
+                refit_full = 'all'
 
-        if refit_full:
-            learner.refit_ensemble_full()
+        if refit_full is not False:
+            predictor.refit_full(model=refit_full)
             if set_best_to_refit_full:
-                trainer = learner.load_trainer()
-                if trainer.model_best in trainer.model_full_dict.keys():
-                    trainer.model_best = trainer.model_full_dict[trainer.model_best]
+                trainer = predictor._trainer
+                trainer_model_best = trainer.get_model_best()
+                if trainer_model_best in trainer.model_full_dict.keys():
+                    trainer.model_best = trainer.model_full_dict[trainer_model_best]
                     # Note: model_best will be overwritten if additional training is done with new models, since model_best will have validation score of None and any new model will have a better validation score.
                     # This has the side-effect of having the possibility of model_best being overwritten by a worse model than the original model_best.
                     trainer.save()
                 else:
-                    logger.warning(f'Best model ({trainer.model_best}) is not present in refit_full dictionary. AutoGluon will default to using {trainer.model_best} for predictions.')
+                    logger.warning(f'Best model ({trainer_model_best}) is not present in refit_full dictionary. AutoGluon will default to using {trainer_model_best} for predictions.')
```

Comment:
"not present in refit_full dictionary" could be clarified by stating causal reasons why it might not be present

Reply:
Addeed